src/components/PostDetails.tsx

```diff
@@ -1,105 +1,59 @@
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addComment, deleteComment,
+  fetchComments,
+} from '../features/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const post
+    = useAppSelector(state => state.selectedPost.selectedPost);
+  const dispatch = useAppDispatch();
+  const { items, loaded, hasError } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
     setVisible(false);
-    setError(false);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post?.id !== undefined) {
+      dispatch(fetchComments(post.id));
     }
-  };
-
-  useEffect(() => {
-    loadComments();
-  }, []);
+  }
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  useEffect(loadComments, [post?.id]);
 
-  const addComment = async ({ name, email, body }: CommentData) => {
-    try {
+  const addCommen = async ({ name, email, body }: CommentData) => {
+    if (post?.id !== undefined) {
       const newComment = await commentsApi.createComment({
         name,
         email,
         body,
         postId: post.id,
       });
 
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(addComment(newComment));
     }
   };
 
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
-
+  const deleteCommen = async (commentId: number) => {
+    dispatch(deleteComment(commentId));
```

Comment:
it's a bad idea to create names with typos because you already have such a name in the code, describe it with synonyms
- handleDeleteComment
- removeComment
- your variant ...